File: Day1-Arrays/2_PascalsTriangle.cpp
#include <bits/stdc++.h> 

vector<vector<long long int>> printPascal(int n) 
{
    vector<vector<long long>> pascal(n, vector<long long>(0));
    for(int i=1; i<=n; i++)
    {
        pascal[i-1] = vector<long long>(i, 1L);
        for(int j=1; j<i-1; j++)
            pascal[i-1][j] = pascal[i-2][j-1] + pascal[i-2][j];
    }
        
    return pascal;
}

File: Day1-Arrays/2_PascalsTriangle.java
import java.util.ArrayList;

class Solution {

	public static ArrayList<ArrayList<Long>> printPascal(int n) {

    	ArrayList<ArrayList<Long>> pascal = new ArrayList<>(n);
        
        for(int i=0; i<n; i++) {
        	
            pascal.add(new ArrayList<Long>(i+1));
        	
            for(int j=0; j<i+1; j++) {
        		if(j == 0 || j == i)
        			pascal.get(i).add(1L);
        		else
        			pascal.get(i).add(pascal.get(i-1).get(j) + pascal.get(i-1).get(j-1));
        	}
        }

        return pascal;

	}
}